src/common/main.lua

```diff
@@ -69,11 +70,87 @@ main.init = function()
 			["Message-Id"] = msg.Id,
 		})
 		-- Send HyperBEAM patch message with updated ACL that has the ant removed
+		-- only send if the acl is not empty, as we will nuke the entire ACL from the patch device if it is empty
+		if #utils.keys(acl) > 0 then
+			ao.send({
+				device = "patch@1.0",
+				cache = { acl = acl },
+			})
+		end
+	end)
 
-		ao.send({
-			device = "patch@1.0",
-			cache = { acl = acl },
-		})
+	utils.createActionHandler(ActionMap.BatchUnregister, function(msg)
+		assert(msg.From == Owner, "Only ANT Registry owner can batch unregister")
+		local antIds = json.decode(msg.Data)
+		assert(
+			type(antIds) == "table" and #antIds > 0 and utils.validateArweaveId(antIds[1]),
+			"msg.Data must be a table of antIds, recieved " .. type(antIds)
+		)
+
+		--- for the patch device
+		local patchAcl = {}
+		--- At the end of the loop, if there are any errors, we will send a notice to the owner with the error
+		--- messages for each antId that had an error.
+		local antIdErrorMap = {}
+		--- we do this to skip over duplicate antIds in the array
+		--- these would error out because we already unregistered them
+		local visitedAntIds = {}
```

Comment:
nit: consider doing prework to create a unique list of VALID ids, throwing out any invalid ones or duplicates, THEN operate on them

Reply:
https://github.com/ar-io/ar-io-ant-registry-process/pull/40/commits/c6a8ca80661b1cccd5a857d97aa851a580bd6cb9